syntax = "proto3";

// message: define message structure.
// refer to https://developers.google.com/protocol-buffers/docs/proto3#scalar for other data types
message ImageUpload {
    bytes image_data = 1; // serial number, increment from 1 to n
}

message ClassificationResult {
    int32 cls = 1;
}

// service nickname
// rpc: input output configuration
service ImageClassification {
    rpc ImageClassificationRequestHandler(ImageUpload) returns (ClassificationResult);
}